src/python/ensembl/genes/metrics/check_busco_score.py

```diff
@@ -0,0 +1,117 @@
+    and return it as a float.
+    If the format is invalid, raise a ValueError.
+    :param busco_str: The BUSCO string to parse.
+    :return: The completeness percentage as a float.
+    """
+    match = re.search(r"C:([\d.]+)%", busco_str)
+    if match:
+        return float(match.group(1))
+    raise ValueError(f"Invalid BUSCO format: {busco_str}")
+
+
+def evaluate_busco(genome_json_path: str, protein_json_path: str) -> bool:
+    """Evaluate BUSCO scores from genome and protein JSON files.
+    This function loads the BUSCO JSON files for genome and protein,
+    extracts the completeness scores, and compares them to determine
+    if the protein BUSCO score is significantly higher than the genome BUSCO score.
+    The criteria for significance are:
+    - If the protein BUSCO score is >= 70%, it is considered significant.
+    - If the protein BUSCO score is between 50% and 70%, it is considered significant
+      if the difference between the protein and genome BUSCO scores is >= 10%.
+    If the genome or protein BUSCO JSON files cannot be loaded, an error message is printed
+    and the function returns False.
+
+    Args:
+        genome_json_path (str): _path to the genome BUSCO JSON file_
+        protein_json_path (str): _path to the protein BUSCO JSON file_
+
+    Raises:
+        KeyError: If the expected '.busco' key is not found in either JSON file.
+
+    Returns:
+        bool: True if the protein BUSCO score is significantly higher than the genome BUSCO score,
+    """
+    try:
+        with open(genome_json_path) as g_file:
+            genome_data = json.load(g_file)
+    except Exception as e: # pylint: disable=broad-except
+        print(f"Failed to load genome BUSCO JSON: {e}")
+        sys.exit(42)
+
+    try:
+        with open(protein_json_path) as p_file:
+            protein_data = json.load(p_file)
+    except Exception as e: # pylint: disable=broad-except
+        print(f"Failed to load protein BUSCO JSON: {e}")
+        sys.exit(42)
+
+    genome_busco_key = next((k for k in genome_data if k.endswith(".busco")), None)
+    protein_busco_key = next((k for k in protein_data if k.endswith(".busco")), None)
+
+    if not genome_busco_key or not protein_busco_key:
+        raise KeyError("Missing '.busco' key in one of the JSON files")
+
+    genome_busco_score = extract_completeness(genome_data[genome_busco_key])
+    protein_busco_score = extract_completeness(protein_data[protein_busco_key])
+
+    print(f"Genome BUSCO completeness: {genome_busco_score}%")
+    print(f"Protein BUSCO completeness: {protein_busco_score}%")
+
+    if protein_busco_score >= 70:
+        print(f"Protein BUSCO completeness {protein_busco_score}% is above threshold (70%)")
+        sys.exit(0)
+    if 50 < protein_busco_score < 70:
+        difference = abs(protein_busco_score - genome_busco_score)
+        print(f"Difference (protein - genome): {difference:.2f}%")
+        if difference >= 10:
```

Comment:
Maybe these should be made into optional arguments?

Reply:
see commit 24c5223..9d0c49f